### PR TITLE
Static Analyses & CI Improvements

### PR DESCRIPTION
**Pull request by timbrinded**
Adds the following:
- [X] Code formatting scripts and CI checks for typescript tests
- [X] ESLint scripts and CI checks for typescript tests
- [x] mold linker to improve rust build times
- [x] sccache to improve rust build times

### CODE REVIEW

File: test/scripts/deriveTestIds.ts [hunk 1]
```diff
@@ -19,106 +19,108 @@ import yargs from "yargs";
...
+            }
+            fileCount++;
+        }
     }
-  }
 }
 
 function hasSpecialCharacters(filename: string): boolean {
-  return /[ \t!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(filename);
+    return /[ \t!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]+/.test(filename);
```
Comment (tmpolaczyk) [inline test/scripts/deriveTestIds.ts:115]: This regex has changed, was this prettier or was it you?

Comment (timbrinded) [reply test/scripts/deriveTestIds.ts:115]: ESLint told me that the escaping was redundant in this example

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Typescript directories
+**/node_modules
+**/.yarn
+test/tsconfig.json
+
+# Spec/Wasm build directory
+**/build/
+
+# Generated test specs
+/tests/moonbeam-test-specs/*.json
```
Comment (tmpolaczyk) [inline .prettierignore:17]: Probably not needed

**Review by tmpolaczyk — APPROVED** [review]

File: .prettierignore (hunk 2 above)
Comment (timbrinded) [reply .prettierignore]: removed

**Review by girazoki — APPROVED** [review]

**Pull request merged**